src/NCalc.Core/ExpressionOptions.cs

```diff
@@ -62,5 +62,10 @@ public enum ExpressionOptions
     /// <summary>
     /// Check for arithmetic binary operation overflow
     /// </summary>
-    OverflowProtection = 1 << 10
+    OverflowProtection = 1 << 10,
+
+    /// <summary>
+    /// Concat values as strings instead of arithmetic addition
+    /// </summary>
```

Comment:
What about `Prioritizes string concatenation over arithmetic addition when one of the values ​​is a string and the other is a number.`?

Reply:
It makes sense, we just need to decide which one is better

Reply:
> which one is better

Sorry I did not understand, what comment is better or behavior? I think the non breaking behavior is better

Reply:
Unfortunately, there is no non-breaking changes. I've updated the PR to match this logic:

```
'L' + 'eo' -> Always concat regardless of StringConcat
1 + '2' -> Same behavior as old versions with StringConcat off and '12' with StringConcat on
'1' + '2' -> 3 with StringConcat off and '12' with StringConcat on
```

Reply:
Ooops, seems first case is not working like in a sample, will update the PR again

Reply:
Should be ok now

Reply:
> Unfortunately, there is no non-breaking changes. I've updated the PR to match this logic:
> 
> ```
> 'L' + 'eo' -> Always concat regardless of StringConcat
> 1 + '2' -> Same behavior as old versions with StringConcat off and '12' with StringConcat on
> '1' + '2' -> 3 with StringConcat off and '12' with StringConcat on
> ```

But what is breaking in this logic?

Reply:
The only breaking case is '1'+2, without ExpressionOptions.StringConcat it will be 3, but in current version the results will be '12'

Reply:
Sorry for the late response, I was out of town. I think this breaking change makes sense. Let's merge?

Reply:
No problem, just updated the docs to match current logic. If you think it's ok, we can merge it